sdk/src/main/java/io/radar/sdk/RadarVerificationManager.kt

```diff
@@ -1,29 +1,40 @@
 package io.radar.sdk
 
 import android.content.Context
-import com.google.android.play.core.integrity.IntegrityManagerFactory
-import com.google.android.play.core.integrity.IntegrityTokenRequest
+import android.location.Location
+import android.os.Build
+import android.util.Log
+import com.google.android.play.core.integrity.StandardIntegrityManager
+import com.google.android.play.core.integrity.StandardIntegrityManager.StandardIntegrityToken
+import com.google.android.play.core.integrity.StandardIntegrityManager.StandardIntegrityTokenRequest
+import com.google.android.gms.tasks.Task;
 
 internal class RadarVerificationManager(
     private val context: Context,
     private val logger: RadarLogger,
 ) {
 
-    fun getIntegrityToken(googleCloudProjectNumber: Long?, nonce: String?, block: (integrityToken: String?, integrityException: String?) -> Unit) {
-        logger.d("Requesting integrity token")
-
-        if (googleCloudProjectNumber == null) {
-            val integrityException = "Missing Google Cloud project number"
-
-            logger.d(integrityException)
-
-            block(null, integrityException)
-
-            return
+    fun getRequestHash(location: Location, nonce: String): String {
+        val stringBuffer = StringBuilder()
+        // build a string of installId, latitude, longitude, mocked, nonce, sharing
+        stringBuffer.append(RadarSettings.getInstallId(this.context))
+        stringBuffer.append(location.latitude)
+        stringBuffer.append(location.longitude)
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            //Note(travis): throw an error if SDK isn't JELLY_BEAN_MR2?
+            //by now we can be sure that the SDK is high enough, but IDE generates error
+            stringBuffer.append(location.isFromMockProvider)
         }
```

Comment:
If I'm understanding the interplay between the server and this here, if we don't match the conditional, then we're guaranteed to never match the request hashes because this request hash will be missing one value — is that correct?

Might we want to have an:
```
 } else {
 stringBuffer.append(true) // default to isFromMockProvider: true if device is on a lower SDK version
 }
 ```

Reply:
I guess your comment claims "by now we can be sure that the SDK is high enough, but IDE generates error" so that else would be unnecessary -- can you please explain how do we know the SDK is high enough?

Reply:
I'm a little confused myself! 

I was certain that Nick had a one-liner that short-circuited the `trackVerified` call if the API version was too low. I looked through old versions of `Radar.kt` and `RadarVerificationManager.kt` and can't find it. I guess it's gone. 

The Standard Requests require API level 21, which is `LOLLIPOP`, so really, this line of code shouldn't be reachable if the build version isn't at least `LOLLIPOP`.   `location.isFromMockProvider` is available in `KITKAT`, which is level 19. 

The `if` statement here just prevents the IDE from complaining and generating a compilation error. Would it make more sense for `getRequestHash` to throw an error if `Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP`?

The bigger question is what happens when an app tries to call `trackVerified` and doesn't have API level 21+?  What would happen in the current version?

Reply:
I'm thinking of the iOS impl for [app attest](https://github.com/radarlabs/radar-sdk-ios/blob/34b4ed339487c3a58709bf35b4806ceac5647572/RadarSDK/RadarVerificationManager.m#L35)